### ISSUE TITLE
Permission error during self update

### ISSUE THREAD

**Issue by paulriddle** [Jguer/yay#1032]
#### Affected Version

All versions are affected. I'm on
```
yay v9.2.1 - libalpm v11.0.3
```
#### Issue

During regular update yay tries to update itself and fails.

#### Steps to reproduce

Use not the latest version of yay and run `yay`.

#### Output

I'm not attaching the whole output. Basically it just repeats the `rm` error for all files in `~/.cache/yay/yay/src/yay-9.2.1/.go/pkg/mod/`.
```
==> Making package: yay 9.3.1-1 (Mon 23 Sep 2019 09:34:01 PM +05)
==> Retrieving sources...
  -> Found yay-9.3.1.tar.gz
==> Validating source files with sha1sums...
    yay-9.3.1.tar.gz ... Passed
==> Cleaning up...
rm: cannot remove '~/.cache/yay/yay/src/yay-9.2.1/.go/pkg/mod/*': Permission denied
```

**Comment by paulriddle** [comment]
In about 6 hours I can. Right now I'm on different computer with yay 9.3.1. I don't remember getting an error on this computer and it doesn't even have the `.go` directory. Here is what I have on computer with no issue:

```
~> ls -al ~/.cache/yay/yay/
total 3940
drwxr-xr-x  3 fixnum fixnum     130 Sep 23 10:24 ./
drwxr-xr-x 14 fixnum fixnum     276 Sep 23 10:23 ../
drwxr-xr-x  8 fixnum fixnum     180 Sep 23 10:24 .git/
-rw-r--r--  1 fixnum fixnum      35 Sep 23 10:23 .gitignore
-rw-r--r--  1 fixnum fixnum     694 Sep 23 10:23 PKGBUILD
-rw-r--r--  1 fixnum fixnum     490 Sep 23 10:23 .SRCINFO
-rw-r--r--  1 fixnum fixnum 3664949 Sep 23 10:24 yay-9.3.1-1-x86_64.pkg.tar.lzo
-rw-r--r--  1 fixnum fixnum  353866 Sep 23 10:24 yay-9.3.1.tar.gz
```

It's completely different from what I have on computer with the issue.

I did take a look at what's in there and remembered that everything inside `~/.cache/yay/yay/src/yay-9.2.1/.go` had `-r--r--r--` permissions. I also remember that it had these packages inside https://github.com/Jguer/yay/tree/master/vendor/github.com.

I'm sure I can just wipe out the cache or even reinstall yay and everything will be fine, but it's not clear why I have unwritable `.go` there in the first place.

**Comment by paulriddle** [comment]
Here goes
```
~> ll ~/.cache/yay/yay/
total 352K
-rw-r--r-- 1 interpreter wheel  694 Sep 21 00:20 PKGBUILD
drwxr-xr-x 3 interpreter wheel   23 Jun 20 19:02 src/
-rw-r--r-- 1 interpreter wheel 346K Sep 23 21:09 yay-9.3.1.tar.gz
~> ll ~/.cache/yay/yay/src
total 0
drwxr-xr-x 3 interpreter wheel 17 Jun 20 19:02 yay-9.2.1/
~> ll ~/.cache/yay/yay/src/yay-9.2.1/
total 0
~> ll ~/.cache/yay/yay/src/yay-9.2.1/.go
total 0
drwxr-xr-x 3 interpreter wheel 17 Jun 20 19:02 pkg/
~> ll ~/.cache/yay/yay/src/yay-9.2.1/.go/pkg/
total 0
drwxr-xr-x 3 interpreter wheel 24 Jun 20 19:02 mod/
~> ll ~/.cache/yay/yay/src/yay-9.2.1/.go/pkg/mod/
total 0
drwxr-xr-x 5 interpreter wheel 59 Jun 20 19:02 github.com/
~> ll ~/.cache/yay/yay/src/yay-9.2.1/.go/pkg/mod/github.com/
total 0
drwxr-xr-x 3 interpreter wheel 56 Jun 20 19:02 '!jguer'/
drwxr-xr-x 3 interpreter wheel 52 Jun 20 19:02  mikkeloscar/
drwxr-xr-x 4 interpreter wheel 87 Jun 20 19:02 '!morganamilo'/
~> ll ~/.cache/yay/yay/src/yay-9.2.1/.go/pkg/mod/github.com/!jguer/
total 4.0K
dr-x------ 4 interpreter wheel 4.0K Jun 20 19:02 'go-alpm@v0.0.0-20190302211415-9c82d5170ce0'/
~> ll ~/.cache/yay/yay/src/yay-9.2.1/.go/pkg/mod/github.com/!jguer/go-alpm@v0.0.0-20190302211415-9c82d5170ce0/
total 88K
-r--r--r-- 1 interpreter wheel 1.1K Jun 20 19:02 alpm.go
-r--r--r-- 1 interpreter wheel 1.3K Jun 20 19:02 alpm_test.go
-r--r--r-- 1 interpreter wheel  952 Jun 20 19:02 callbacks.c
-r--r--r-- 1 interpreter wheel 1.3K Jun 20 19:02 callbacks.go
-r--r--r-- 1 interpreter wheel 3.8K Jun 20 19:02 db.go
-r--r--r-- 1 interpreter wheel 1.2K Jun 20 19:02 deps.go
-r--r--r-- 1 interpreter wheel 2.7K Jun 20 19:02 enums.go
-r--r--r-- 1 interpreter wheel  486 Jun 20 19:02 error.go
dr-xr-xr-x 2 interpreter wheel   94 Jun 20 19:02 examples/
-r--r--r-- 1 interpreter wheel  15K Jun 20 19:02 handle.go
-r--r--r-- 1 interpreter wheel 1.1K Jun 20 19:02 LICENSE
-r--r--r-- 1 interpreter wheel 7.9K Jun 20 19:02 package.go
-r--r--r-- 1 interpreter wheel 2.6K Jun 20 19:02 package_test.go
-r--r--r-- 1 interpreter wheel  667 Jun 20 19:02 README.md
-r--r--r-- 1 interpreter wheel  760 Jun 20 19:02 sync.go
dr-xr-xr-x 2 interpreter wheel   38 Jun 20 19:02 testing/
-r--r--r-- 1 interpreter wheel  958 Jun 20 19:02 trans.go
-r--r--r-- 1 interpreter wheel 6.2K Jun 20 19:02 types.go
-r--r--r-- 1 interpreter wheel  395 Jun 20 19:02 version.go
~>
```

**Comment by Jguer** [comment]
That's weird, I confess I've never seen that one. Are you using gcc-go by any chance?

**Comment by paulriddle** [comment]
No, regular go, nothing weird. I didn't even monkey with the initial install of yay, it was pretty standard. During my use of yay I run nothing but `yay` and `yay -Syu --devel` (a couple of times).

Anyway, either I did something weird and forgot about it, or there is a bug somewhere. It's clear how to solve this one for me. Close this?

**Comment by Morganamilo** [comment]
I'ts `go` setting the directory to read only so you don't accidentally modify it or something like that. I've had the same thing happen to other go based pkgbuilds.

**Comment by Jguer** [comment]
Using go-modules but with the $GOPATH set to the $pwd/.go directory right? I guess moving to go modules only and eliminating vendor will fix this

**Comment by Morganamilo** [comment]
>Using go-modules but with the $GOPATH set to the $pwd/.go directory right? I guess moving to go modules only and eliminating vendor will fix this

For the other packages, I have no idea how they do things. As for my setup all the go variables are unset so assume defaults.

Actually, testing it myself. I can not reproduce this with `makepkg -c`, while I can with yay.  So I guess it is something to do with yay and not entirely go's fault.

**Comment by Jguer** [comment]
Fixed in fdcfeba presumably.
Writeup in #1047
